util/journaldb/src/lib.rs

```diff
@@ -53,6 +56,32 @@ pub mod overlaydb;
 /// Export the `JournalDB` trait.
 pub use self::traits::JournalDB;
 
+use archivedb::ArchiveDB;
+impl AsHashDB<KeccakHasher> for ArchiveDB {
+	fn as_hashdb(&self) -> &HashDB<KeccakHasher> { self }
+	fn as_hashdb_mut(&mut self) -> &mut HashDB<KeccakHasher> { self }
+}
+use earlymergedb::EarlyMergeDB;
```

Comment:
style: i'm not a huge fan of these interleaved `use` and `impl` blocks. maybe we could do this all in an embedded submodule?